api/server/v1/tx.go

```diff
@@ -21,24 +21,70 @@ import (
 )
 
 var (
-	methodPrefix             = "/tigrisdata.v1.Tigris/"
-	insert                   = methodPrefix + "Insert"
-	replace                  = methodPrefix + "Replace"
-	update                   = methodPrefix + "Update"
-	delete                   = methodPrefix + "Delete"
-	read                     = methodPrefix + "Read"
-	commitTransaction        = methodPrefix + "CommitTransaction"
-	rollbackTransaction      = methodPrefix + "RollbackTransaction"
-	dropCollection           = methodPrefix + "DropCollection"
-	listCollection           = methodPrefix + "ListCollections"
+	methodPrefix = "/tigrisdata.v1.Tigris/"
+
+	insert  = methodPrefix + "Insert"
+	replace = methodPrefix + "Replace"
+	update  = methodPrefix + "Update"
+	deleteR = methodPrefix + "Delete"
+	read    = methodPrefix + "Read"
+
+	search = methodPrefix + "Search"
+
+	//publish   = methodPrefix + "Publish"
+	subscribe = methodPrefix + "Subscribe"
+
+	events = methodPrefix + "Events"
+
+	//beginTransaction    = methodPrefix + "BeginTransaction"
+	commitTransaction   = methodPrefix + "CommitTransaction"
+	rollbackTransaction = methodPrefix + "RollbackTransaction"
+
 	createOrUpdateCollection = methodPrefix + "CreateOrUpdateCollection"
+	dropCollection           = methodPrefix + "DropCollection"
+
+	//createDatabase = methodPrefix + "CreateDatabase"
+	//dropDatabase   = methodPrefix + "DropDatabase"
+
+	listDatabases   = methodPrefix + "ListDatabases"
+	listCollections = methodPrefix + "ListCollections"
+
+	describeDatabase   = methodPrefix + "DescribeDatabase"
+	describeCollection = methodPrefix + "DescribeCollection"
 )
 
+func isRead(name string) bool {
+	switch name {
+	case read, events, search, subscribe:
+		return true
+	case listCollections, listDatabases:
+		return true
+	case describeCollection, describeDatabase:
+		return true
+	default:
+		return false
+	}
+}
+
+func isWrite(name string) bool {
```

Comment:
probably we can move these methods in `server/request/request.go` in `RequestMetadata`. wdyt?